optuna/trial.py

```diff
@@ -401,6 +401,18 @@ def _suggest(self, name, distribution):
         param_value = distribution.to_external_repr(param_value_in_internal_repr)
         return param_value
 
+    @property
+    def number(self):
+        # type: () -> int
+        """Return trial's number which is consecutive and unique in a study.
+
+        Returns:
+            A trial number.
+        """
+
+        # For studies created by optuna<=0.5.0, return trial_id if the trial number is not found.
+        return self.system_attrs.get('number', self.trial_id)
```

Comment:
Because to return `self.trial_id` breaks the postcondition (i.e., "return a number **which is consecutive**") of the method, I think it is better to re-calculate the trial number in that case.
But, I want to hear the opinions of other members.

Reply:
> re-calculate the trial number

That makes sense. I'll implement it. Thanks!